src/ngraph/op/fused/scatter_nd.cpp

```diff
@@ -0,0 +1,109 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/fused/scatter_nd.hpp"
+#include "ngraph/node.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/scatter_nd_add.hpp"
+#include "ngraph/op/select.hpp"
+#include "ngraph/shape.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static int DATA = 0;
+static int INDICES = 1;
+static int UPDATES = 2;
+
+constexpr NodeTypeInfo op::ScatterND::type_info;
+
+shared_ptr<Node> op::ScatterND::copy_with_new_args(const NodeVector& new_args) const
+{
+    check_new_args_count(this, new_args);
+    return make_shared<ScatterND>(new_args.at(DATA), new_args.at(INDICES), new_args.at(UPDATES));
+}
+
+void op::ScatterND::validate_and_infer_types()
```

Comment:
When you switch to `FusedOp` as a base class, this method should be renamed to `pre_validate_and_infer_types`

Reply:
Done